### PR TITLE
Adapt test-assert.js test from Node.js to IoT.js.

### PR DESCRIPTION
**Pull request by zherczeg**

### CODE REVIEW

**Comment by daeyeon** [comment]
@zherczeg Kindly let me have your thought on my suggestion.

First of all, we don't need to add the unit test for `assert.ok` since it's not described in the document. (not supported.) 

Second, we can find many node.js tests that use `assert.ok`. So, for using them with minimal changes, we could extend `assert` in common.js like below.

`test.js`
```
var common = require('./common');
var assert = require('assert');

// Assuming the below is a test from node.js, it doesn't work since 
// IoT.js doesn't support assert.ok. So, assert.ok is extended in common.js.

test(function serverTimeout(cb) {
    var server = https.createServer(....);
    server.listen(0, common.mustCall(function () {
        ...
        assert.ok(s instanceof https.Server); // <-- This works now.
        ...
    }));
});
```

`common.js`
```
var assert = require('assert');

assert.ok = function (value, message) {
    assert.equal(!!value, true, message);
}
```

File: test/node/parallel/test-assert.js [hunk 1]
```diff
@@ -0,0 +1,730 @@
...
+// Test starts here
+
+function makeBlock(f) {
+  var args = Array.prototype.slice.call(arguments, 1);
+  return function() {
+    return f.apply(this, args);
+  };
+}
+
+assert.ok(a.AssertionError.prototype instanceof Error,
```
Comment (daeyeon) [inline test/node/parallel/test-assert.js]: we don't need to add the unit test for assert.ok

**Comment by zherczeg** [comment]
Could you review this patch?

**Comment by daeyeon** [comment]
This a kind of skeleton for #853 and will be incrementally updated.

**Comment by zherczeg** [comment]
@daeyeon could you check this again?

**Comment by yichoi** [comment]
LGTM

**Pull request merged**